src/psyclone/psyGen.py

```diff
@@ -6777,6 +6883,107 @@ def _ignore_handler(self, *_):
+        self.process_nodes(parent=loop_body, nodes=node.content[1:-1],
+                           nodes_parent=node)
+
+    def _do_construct_handler(self, node, parent):
+        '''
+        Transforms a fparser2 Do Construct into its PSyIR representation.
+
+        :param node: node in fparser2 tree.
+        :type node: \
+            :py:class:`fparser.two.Fortran2003.Block_Nonlabel_Do_Construct`
+        :param parent: Parent node of the PSyIR node we are constructing.
+        :type parent: :py:class:`psyclone.psyGen.Node`
+
+        :returns: PSyIR representation of node
+        :rtype: :py:class:`psyclone.psyGen.Loop`
+
+        :raises InternalError: If the fparser2 tree has an unexpected \
+            structure.
+        '''
+        from fparser.two.utils import walk_ast
+        from fparser.two import Fortran2003
+        ctrl = walk_ast(node.content, [Fortran2003.Loop_Control])
+        if not ctrl:
+            raise InternalError(
+                "Unrecognised form of DO loop - failed to find Loop_Control "
+                "element in the node '{0}'.".format(str(node)))
+        if ctrl[0].items[0]:
+            # If this is a DO WHILE then the first element of items will not
+            # be None. (See `fparser.two.Fortran2003.Loop_Control`.)
+            # TODO #359 DO WHILE's are currently just put into CodeBlocks
+            # rather than being properly described in the PSyIR.
+            raise NotImplementedError()
+
+        # Second element of items member of Loop Control is itself a tuple
+        # containing:
+        #   Loop variable, [start value expression, end value expression, step
+        #   expression]
+        # Loop variable will be an instance of Fortran2003.Name
+        loop_var = str(ctrl[0].items[1][0])
+        variable_name = str(loop_var)
+        loop = self._create_loop(parent, variable_name)
+        loop._ast = node
+
+        # Get the loop limits. These are given in a list which is the second
+        # element of a tuple which is itself the second element of the items
+        # tuple:
+        # (None, (Name('jk'), [Int_Literal_Constant('1', None), Name('jpk'),
+        #                      Int_Literal_Constant('1', None)]), None)
+        limits_list = ctrl[0].items[1][1]
+
+        # Start expression child
+        self.process_nodes(parent=loop, nodes=[limits_list[0]],
+                           nodes_parent=ctrl)
+
+        # Stop expression child
+        self.process_nodes(parent=loop, nodes=[limits_list[1]],
+                           nodes_parent=ctrl)
+
+        # Step expression child
+        if len(limits_list) == 3:
+            self.process_nodes(parent=loop, nodes=[limits_list[2]],
+                               nodes_parent=ctrl)
+        else:
+            # Default loop increment is 1
+            default_step = Literal("1", parent=loop)
+            loop.addchild(default_step)
+
+        # Process loop body (ignore 'do' and 'end do' statements with [1:-1])
```

Comment:
There's no `[1:-1]` in the code here.

Reply:
Ups moved the comment line to where the [1:-1] is